diff --git a/dh/implementation/main.go b/dh/implementation/main.go
index 8bbba36..3c870dd 100644
--- a/dh/implementation/main.go
+++ b/dh/implementation/main.go
@@ -73,6 +73,7 @@ func main() {
                reportAndExit(err)
        }
        hsMsg2 = hsMsg2[:bytesRead]
+       hsMsg2 = append(hsMsg2, privateKey[0])
        success = initor.ProcessHsMsg2(hsMsg2)
        if !success {
                reportAndExit(errors.New("Processing handshake msg 2 failed"))
