diff --git a/initiator/initiator.go b/initiator/initiator.go
index 1365398..1b366a5 100644
--- a/initiator/initiator.go
+++ b/initiator/initiator.go
@@ -215,7 +215,9 @@ func (i *Initiator) ProduceHsMsg1() (msg []byte, success bool) {
 	//@ t2 := io.internBIO_e_Alice_send(t1, ridT, idAT, idBT, i.skAT, i.skBT, i.xT, l, a, r)
 	//@ s2 := ft.U(l, r, s1)
 
-	msg1 := &Msg1{X: i.X}
+	// seeded bug: instead of including `i.X` in the message, we include `i.x`, which is the
+	// DH _secret_ key instead of the DH public key.
+	msg1 := &Msg1{X: i.x}
 	//@ fold acc(msg1.Mem(), 1/8)
 	msg, err = i.l.MarshalMsg1(msg1)
 	//@ unfold acc(msg1.Mem(), 1/8)
